### DISCUSSION TITLE
Play Billing v5

### DISCUSSION THREAD

**Discussion by marchbold** [airsdk/Adobe-Runtime-Support#2165] [General]
Hi all,

You may have received a warning from Google Play recently along the lines of:

> We've detected that this app is using an old version of Google Play Billing. By November 1, 2022, all app updates must use Billing Library version 4 or newer. Update to Billing Library 4 before this date.

If you are using our InAppBilling extension and have some time to do some testing, our latest beta release with v5 of Play Billing is available in the [repository](https://github.com/distriqt/ANE-InAppBilling/issues/482#issuecomment-1253242955). 

The deadline for apps to update to the new release is approaching so we would appreciate any feedback on the new API for subscriptions and getting as many testers on it as possible before releasing.

Cheers, 
Michael

**Comment by hydroper** [comment]


**Reply by marchbold** [reply]
Yes, this is for our subscribed users only.

**Comment by Michoko92** [comment]
Hi, do you mean that the implementation for regular entitlements IAP did not change, so it is pretty safe to migrate if we don't use subscriptions?

**Reply by marchbold** [reply]
Hi, the API is the same for regular in app purchases so there shouldn't be any changes for consumable/non-consumable products. 

However we have rewritten a lot internally so would appreciate testing of all aspects.

**Comment by Shaunmax** [comment]
Hello @marchbold 

I have done the basic testing for IAP purchases on android & did not face any issue.
Could you please tell us what exactly are the changes in it. Next week we will be testing the same on iOS.

Thanks

**Reply by marchbold** [reply]
It's a major migration from v3.0.3 to v5.0.0 of the play billing library. If you would like to see details there is a change log here: https://developer.android.com/google/play/billing/release-notes

Basically a lot of operations have changed from synchronous to asynchronous and there is a major change to subscriptions, from simple products with prices and subscriptions (and some discounts) to having multiple ways to purchase a subscription (called offers containing phases). As part of this a lot of details around products changed including the data type for products (all handled internally apart from the new api for subscription offers and phases).  

Currently I don't believe we have covered *all* of the new subscription functionality, but we have covered the core updates and functionality.

**Reply by Shaunmax** [reply]
so, its just internal changes! 
Basic purchase flow everything should remain same right ? 
I will also make sure to check the analytics and data parts as well.

thanks

**Reply by marchbold** [reply]
As long as you aren't using subscriptions, it should all be internal changes yes (but a lot of them :) )

**Reply by marchbold** [reply]
Just a small note, no code changes but you do have to change the manifest tag for the billing library version from 3.0.3 to 5.0.0 ;)

**Reply by Shaunmax** [reply]
I believe this will be taken care of by apm when the final ane is pushed live, right?

**Reply by marchbold** [reply]
Yes definitely when we release the package it will correctly set the manifest additions.

**Comment by marchbold** [comment]
The update to [InAppBilling](https://airnativeextensions.com/extension/com.distriqt.InAppBilling) including v5 of the play billing library has now been released! Appreciate everyone who helped test out the update.

Don't forget the deadline for updating to #GooglePlay Billing library v5.0.0 is on the 1st of November. 

You will need to make sure you have migrated to the latest version of the #InAppBilling extension to submit updates after this date.

https://docs.airnativeextensions.com/news/2022/10/07/inappbilling-v14-update

**Comment by idanasher** [comment]
I am also got this warning and want tp upgrade,

question: which (stable) air sdk version do you recommend to use ?
